remix-analyzer/src/analysis/modules/stringBytesLength.js

```diff
@@ -0,0 +1,34 @@
+var name = 'String Length: '
+var desc = 'Bytes length != String length'
+var categories = require('./categories')
+var common = require('./staticAnalysisCommon')
+
+function stringBytesLength () {
+  this.stringToBytesConversions = []
+  this.bytesLengthChecks = []
+}
+
+stringBytesLength.prototype.visit = function (node) {
+  if (common.isStringToBytesConversion(node)) this.stringToBytesConversions.push(node)
+  else if (common.isBytesLengthCheck(node)) this.bytesLengthChecks.push(node)
+}
+
+stringBytesLength.prototype.report = function (compilationResults) {
+  if (this.stringToBytesConversions.length > 0 && this.bytesLengthChecks.length > 0) {
+    return [{
+      warning: 'Bytes and string length are not the same since strings are assumed to be UTF-8 encoded (according to the ABI defintion) therefore one character is not nessesarily encoded in one byte of data.',
+      location: this.bytesLengthChecks[0],
+      more: 'https://github.com/ethereum/wiki/wiki/Ethereum-Contract-ABI#argument-encoding'
```

Comment:
Please link to the Solidity documentation, that has a much more up to date ABI section. Also could just point to the `string` section instead :)

Reply:
Can you maybe point me to a good section in the documentation I looked for something about utf8 encoding but I found not to much about strings. 
Except for https://solidity.readthedocs.io/en/v0.4.24/frequently-asked-questions.html?highlight=strings#what-are-some-examples-of-basic-string-manipulation-substring-indexof-charat-etc which actually suggests to convert to bytes to check string length. Kind of amazing.

Reply:
https://solidity.readthedocs.io/en/v0.4.24/types.html#dynamically-sized-byte-array

I think the message above I confusing "bytes and string length are not the same". I think it should explain that `.length` does give you the number of bytes occupied in both cases, but strings are supposed to be UTF-8 encoded (however they are usually user supplied) and as a result `.length` in many cases will not be equal to the number of characters.